core/serialization.c

```diff
@@ -115,15 +135,21 @@ Shape *serialization_load_shape(Stream *s,
     stream_free(s);
 
     // shrink box once all blocks were added to update box origin
+    if (doubly_linked_list_node_count(list) == 0) {
+        doubly_linked_list_free(list);
+        list = NULL;
+        cclog_error("[serialization_load_resources] no resources found");
+    }
+    
+    // set fullname if containing a root shape
+    Shape *shape = resources_get_root_shape(list);
     if (shape != NULL) {
-        shape_shrink_box(shape);
```

Comment:
You added this call inside `serialization_load_resources_v6`, but it is now missing for a v5 shape. You could add it in `serialization_v5_load_shape`